src/journal/managed-journal-file.c

```diff
@@ -167,19 +167,31 @@ static void managed_journal_file_set_offline_internal(ManagedJournalFile *f) {
         for (;;) {
                 switch (f->file->offline_state) {
                 case OFFLINE_CANCEL:
-                        if (!__sync_bool_compare_and_swap(&f->file->offline_state, OFFLINE_CANCEL, OFFLINE_DONE))
-                                continue;
-                        return;
+                        {
+                                OfflineState tmp_state = OFFLINE_CANCEL;
+                                if (!__atomic_compare_exchange_n(&f->file->offline_state, &tmp_state, OFFLINE_DONE,
+                                    false, __ATOMIC_SEQ_CST, __ATOMIC_SEQ_CST))
+                                        continue;
+                                return;
+                        }
```

Comment:
we usually would indent this like this:

```c
case OFFLINE_CANCEL: {
        OfflineState tmp_state …
}
case OFFLINE_AGAIN_FROM_SYNCING:…
```

please follow the same logic here